src/core/search/base.h

```diff
@@ -21,24 +25,34 @@ using OwnedFtVector = std::pair<std::unique_ptr<float[]>, size_t /* dimension (s
 
 // Query params represent named parameters for queries supplied via PARAMS.
 struct QueryParams {
+  std::string_view operator[](std::string_view name) const;
+  std::string& operator[](std::string_view k);
+
   size_t Size() const {
     return params.size();
   }
 
-  std::string_view operator[](std::string_view name) const {
-    if (auto it = params.find(name); it != params.end())
-      return it->second;
-    return "";
-  }
+ private:
+  absl::flat_hash_map<std::string, std::string> params;
+};
 
-  std::string& operator[](std::string_view k) {
-    return params[k];
-  }
+struct SortOption {
+  std::string field;
+  bool descending = false;
+};
+
+struct WrappedStrPtr {
+  WrappedStrPtr(const std::string& s);
```

Comment:
yes I really want this: implicit and from const string&

Reply:
add it as a comment then? :)